plugins/auth-node/src/DefaultIdentityClient.ts

```diff
@@ -78,17 +78,32 @@ export class DefaultIdentityClient implements IdentityApi {
       : ['ES256'];
   }
 
-  async getIdentity({ request }: IdentityApiGetIdentityRequest) {
+  async getIdentity({
+    request,
+    optional = false,
+  }: IdentityApiGetIdentityRequest): Promise<
+    BackstageIdentityResponse | undefined
+  > {
+    let id: BackstageIdentityResponse | undefined;
     if (!request.headers.authorization) {
+      if (!optional) {
+        throw new AuthenticationError('Identity was not provided');
```

Comment:
I discussed this with @Rugvip and we realised that this is a breaking change from the current behaviour were we return `undefined` if the header is not set. 
We still think this is a good fix but plugins that treat the token as optional today need to be updated to pass in `optional: true`. So it would be good check/update existing usages accordingly.